L1Trigger/L1THGCal/src/concentrator/HGCalConcentratorAutoEncoderImpl.cc

```diff
@@ -16,15 +19,43 @@ HGCalConcentratorAutoEncoderImpl::HGCalConcentratorAutoEncoderImpl(const edm::Pa
       zeroSuppresionThreshold_(conf.getParameter<double>("zeroSuppresionThreshold")),
       saveEncodedValues_(conf.getParameter<bool>("saveEncodedValues")),
       preserveModuleSum_(conf.getParameter<bool>("preserveModuleSum")) {
-  //construct inverse array, to get U/V for a particular ae output position
+  // find total size of the expected input shape
+  // used for checking the maximum size used in cell Remap
+  nInputs_ = 1;
+  for (const auto& i : encoderShape_) {
+    nInputs_ *= i;
+  }
+
+  // check the size of the inputs shapes
+  if (encoderShape_.size() != encoderTensorDims_) {
+    throw cms::Exception("BadInitialization") << "Encoder input shapes are currently expected to be 4 values";
```

Comment:
Now that `encoderTensorDims_` has been defined, it would be better to use it in the exception message